Start splitting of rhel7 CIS

vojtapolasek opened the pull request:
#### Description:

- create new policy for CIS
- create 3 new profiles based on levels in this policy
- change the "cis" profile to be the level2 server
- implement section 1 of CIS benchmark

#### Rationale:

update of rhel7 cis and split it to allow more granularity

yuumasato reviewed (changes requested):
Found a few issues but it is looking good.

> * create 3 new profiles based on levels in this policy
> 
> * change the "cis" profile to be the level2 server

I like the format of the new profile IDs, and the profile id inconsistency annoys me a bit.

Idea: What if we added  the `cis_server_l2` profile,  and kept the original `cis.profile` as an extension of `cis_server_l2` profile?
The plain `cis` profile would work as a "symlink" for automation compatibility and could have "deprecated" in the title. The deprecated profile could be removed after a certain number of releases.

cipherboy commented on controls/cis_rhel7.yml:
```diff
@@ -0,0 +1,554 @@
+    - l1_workstation
+    automated: yes
+    notes: >-
+      This control covers also 1.6.1.5, set the variable appropriately.
```
JFTR, this is the approach we took with AppArmor policy for the corresponding Ubuntu items (a variable with two values). HTH :-)

yuumasato replied:
I see, thanks for sharing.

We discussed this yesterday and we will go a different path, which is to strongly recommend (by selecting enforcing by default) to use SELinux in enforcing mode, even on L1.
If some one really needs to use permissive mode, they will need to tailor the profile. 
@vojtapolasek Could you reflect this decision in the `notes` section?

yuumasato commented:
> I like the format of the new profile IDs, and the profile id inconsistency annoys me a bit.

So, the annoyance of changing profile IDs in already deployed system is much higher than the having inconsistency IDs.

> Idea: What if we added the `cis_server_l2` profile, and kept the original `cis.profile` as an extension of `cis_server_l2` profile?
> The plain `cis` profile would work as a "symlink" for automation compatibility and could have "deprecated" in the title. The deprecated profile could be removed after a certain number of releases.

The existence of two CIS L2 Server profiles can cause problems and confusions too. There would be duplicates showing up on SCAP Workbnench and Oscap Anaconda Addon.

Given the above, the less disruptive approach is to keep the `cis` profile as the Level 2 Server.

yuumasato commented on products/rhel7/profiles/cis_workstation_l2.profile:
```diff
@@ -0,0 +1,22 @@
+    Red Hat Enterprise Linux 7 Benchmark™, v3.1.1, released 05-21-2021.
+
+    This profile includes Center for Internet Security®
+    Red Hat Enterprise Linux 7 CIS Benchmarks™ content. It represents the configuration profile "Level 2 - Workstation".
```
Nitpick in verbiage.
```suggestion
    Red Hat Enterprise Linux 7 CIS Benchmarks™ content.
```

yuumasato commented on products/rhel7/profiles/cis_workstation_l2.profile:
```diff
@@ -0,0 +1,22 @@
+title: 'CIS Red Hat Enterprise Linux 7 Benchmark: Level 2 - Workstation'
+
+description: |-
+    This profile defines a baseline that aligns to the Center for Internet Security®
+    Red Hat Enterprise Linux 7 Benchmark™, v3.1.1, released 05-21-2021.
```
```suggestion
    This profile defines a baseline that aligns to configuration "Level 2 - Workstation" from the Center for Internet Security®
    Red Hat Enterprise Linux 7 Benchmark™, v3.1.1, released 05-21-2021.
```

yuumasato reviewed (changes requested):
The addition of the last sentence in the profile description seems out of place to me.
I made a suggestion to incorporate it to the first paragraph, where we state the profile alignment to the benchmark.

Other aspects look good to me.

vojtapolasek commented:
Hello @yuumasato  I changed all descriptions. I also removed the second paragraph, I think it only repeated something what is mentioned in the first paragraph.

yuumasato commented:
> Hello @yuumasato I changed all descriptions. I also removed the second paragraph, I think it only repeated something what is mentioned in the first paragraph.

@vojtapolasek The second paragraph was the credit attribution to CIS. I think we should keep it.

vojtapolasek commented:
I hope I got it right this time :-D

yuumasato reviewed (approved):
Looks great to me, thank you @vojtapolasek for starting the split and updated of the CIS profiles.

yuumasato commented:
@vojtapolasek Sorry, I just noticed that the Ansible Roles lint is complaining about colons `:` in the description.

```
The offending line appears to be:

  author: ComplianceAsCode development team
  description: CIS Red Hat Enterprise Linux 7 Benchmark: Level 2 - Server
                                                       ^ here  
```

vojtapolasek commented:
FixedH

alexhaydock commented:
Great to see that this has been merged! I'm aiming to put in some work soon to refactor my #6976 PR to do the same thing but for RHEL 8.
